packages/db-base/lib/tools.js

```diff
@@ -15,32 +16,40 @@ function getControllerDir() {
             /* not found */
         }
     }
+
     // Apparently, checking vs null/undefined may miss the odd case of controllerPath being ""
     // Thus we check for falsyness, which includes failing on an empty path
     if (!controllerPath) {
-        const checkPath = path.normalize(path.join(__dirname, '../..'));
-        const pathParts = checkPath.split(path.sep);
-        while (pathParts.length) {
-            const tryPath = pathParts.join(path.sep);
-            if (fs.existsSync(path.join(tryPath, 'lib/tools.js'))) {
-                controllerPath = tryPath;
-                break;
+        let checkPath = path.join(__dirname, '../..');
+        // Also check in the current check dir (along with iobroker.js-controller subdirs)
+        possibilities.unshift('');
+        outer: while (true) {
```

Comment:
didn't know that this works ... ok

Reply:
I learned about it a while ago. Very handy when you want to skip nested loops :)